solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -935,6 +945,7 @@ public SolrCore(CoreContainer coreContainer, String name, String dataDir, SolrCo
       MDCLoggingContext.setCore(this);
 
       resourceLoader = config.getResourceLoader();
+      resourceLoader.core = this;
```

Comment:
Is this safe? Isn’t SRL shared by many cores?

Reply:
NO, SRL is per core